pages/posts/[slug].tsx

```diff
@@ -98,7 +105,8 @@ const transformContentfulPost = (item: any): IBlogPost => {
     id: item.sys.id,
     author: item.fields.author,
     body: formatBody(item.fields.body) || '',
-    category: item.fields.category,
+    categoryName: item.fields.category?.fields.name || '',
+    categorySlug: item.fields.category?.fields.slug || '',
     datePublished: formatDate(item.sys.createdAt),
     featuredImageUrl: `https:${item.fields.featuredImage?.fields.file.url}` || '',
     slug: item.fields.slug,
```

Comment:
This code should be in _utils/contentful-posts.ts_, the same way you created the contentful fetching methods. This way, all the contentful related code will remain in the same area (utils).